searx/engines/core.py

```diff
@@ -23,92 +23,104 @@
+        # Get title
+        title = result.get('title')
+        if not title:
+            continue
+
+        # Get URL - try different options
         url = None
-        if source.get('urls'):
-            url = source['urls'][0].replace('http://', 'https://', 1)
+        
+        # Try DOI first
+        doi = result.get('doi')
+        if doi:
+            url = f'https://doi.org/{doi}'
 
-        if url is None and source.get('doi'):
+        if url is None and result.get('doi'):
             # use the DOI reference
-            url = 'https://doi.org/' + source['doi']
-
-        if url is None and source.get('downloadUrl'):
-            # use the downloadUrl
-            url = source['downloadUrl']
-
-        if url is None and source.get('identifiers'):
-            # try to find an ark id, see
-            # https://www.wikidata.org/wiki/Property:P8091
-            # and https://en.wikipedia.org/wiki/Archival_Resource_Key
-            arkids = [
-                identifier[5:]  # 5 is the length of "ark:/"
-                for identifier in source.get('identifiers')
-                if isinstance(identifier, str) and identifier.startswith('ark:/')
-            ]
-            if len(arkids) > 0:
-                url = 'https://n2t.net/' + arkids[0]
-
-        if url is None:
+            url = 'https://doi.org/' + str(result['doi'])
+
+        if url is None and result.get('id'):
+            url = 'https://core.ac.uk/works/' + str(result['id'])
+
+        # Then try downloadUrl
+        if not url and result.get('downloadUrl'):
+            url = result['downloadUrl']
+        
+        if not url and result.get('sourceFulltextUrls'):
+            url = result['sourceFulltextUrls']
+
+        # Skip if no URL found
+        if not url:
             continue
 
-        publishedDate = None
-        time = source['publishedDate'] or source['depositedDate']
-        if time:
-            publishedDate = datetime.fromtimestamp(time / 1000)
+        # Published date
+        published_date = None
+
+        if result.get('publishedDate'):
+            try:
+                published_date = datetime.fromisoformat(result['publishedDate'].replace('Z', '+00:00'))
+            except (ValueError, AttributeError):
+                pass
+
+        if not published_date and result.get('depositedDate'):
+            try:
+                published_date = datetime.fromisoformat(result['depositedDate'].replace('Z', '+00:00'))
+            except (ValueError, AttributeError):
+                pass
 
-        # sometimes the 'title' is None / filter None values
-        journals = [j['title'] for j in (source.get('journals') or []) if j['title']]
+        # Handle journals
+        journals = []
+        if result.get('journals'):
+            journals = [j.get('title') for j in result['journals'] if j.get('title')]
 
-        publisher = source['publisher']
+        # Handle publisher
+        publisher = result.get('publisher', '')
```

Comment:
```suggestion
        publisher = result.get('publisher', '').strip("'")
```
The following lines are unnecessary, you can already strip here.